tests/Microsoft.DotNet.XHarness.Apple.Tests/Orchestration/InstallOrchestratorTests.cs

```diff
@@ -0,0 +1,301 @@
+            {
+                ExitCode = 1, // This can fail as this is the first purge of the app before we install it
+                TimedOut = false,
+            });
+
+        var testTarget = new TestTargetOs(TestTarget.Simulator_tvOS, "13.5");
+
+        // Act
+        var result = await _installOrchestrator.OrchestrateInstall(
+            testTarget,
+            null,
+            AppPath,
+            TimeSpan.FromMinutes(30),
+            includeWirelessDevices: true,
+            resetSimulator: true,
+            enableLldb: true,
+            new CancellationToken());
+
+        // Verify
+        Assert.Equal(ExitCode.SUCCESS, result);
+
+        _deviceFinder.Verify(
+            x => x.FindDevice(testTarget, null, It.IsAny<ILog>(), true),
+            Times.Once);
+
+        VerifySimulatorReset(true);
+        VerifySimulatorCleanUp(false); // Install doesn't end with a cleanup so that the app stays behind
+        VerifyDiagnosticData(testTarget);
+
+        _appInstaller.Verify(
+            x => x.InstallApp(_appBundleInformation, testTarget, _simulator.Object, It.IsAny<CancellationToken>()),
+            Times.Once);
+    }
+
+    [Fact]
+    public async Task OrchestrateDeviceInstallationTest()
+    {
+        // Setup
+        _appInstaller
+            .Setup(x => x.InstallApp(_appBundleInformation, It.IsAny<TestTargetOs>(), _device.Object, It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new ProcessExecutionResult
+            {
+                ExitCode = 0,
+                TimedOut = false,
+            });
+
+        _appUninstaller
+            .Setup(x => x.UninstallDeviceApp(_device.Object, s_appIdentifier, It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new ProcessExecutionResult
+            {
+                ExitCode = 1, // This can fail as this is the first purge of the app before we install it
+                TimedOut = false,
+            });
+
+        var testTarget = new TestTargetOs(TestTarget.Device_iOS, "14.2");
+
+        // Act
+        var result = await _installOrchestrator.OrchestrateInstall(
+            testTarget,
+            null,
+            AppPath,
+            TimeSpan.FromMinutes(30),
+            includeWirelessDevices: false,
+            resetSimulator: false,
+            enableLldb: true,
+            new CancellationToken());
+
+        // Verify
+        Assert.Equal(ExitCode.SUCCESS, result);
+
+        VerifySimulatorReset(false);
+        VerifySimulatorCleanUp(false);
+        VerifyDiagnosticData(testTarget);
+
+        _deviceFinder.Verify(
+            x => x.FindDevice(testTarget, null, It.IsAny<ILog>(), false),
+            Times.Once);
+
+        _appInstaller.Verify(
+            x => x.InstallApp(_appBundleInformation, testTarget, _device.Object, It.IsAny<CancellationToken>()),
+            Times.Once);
+    }
+
+    [Fact]
+    public async Task OrchestrateFailedDeviceInstallationTest()
```

Comment:
The test methods have identical or very similar structure and are quite copy-paste-ish. Would it be possible to refactor this?

Reply:
I factored out common code for Mock creation and verification of results in the test base class, such as:
```csharp
        VerifySimulatorReset(false);
        VerifySimulatorCleanUp(false);
```

Check [here](https://github.com/dotnet/xharness/blob/e74b9e0b5e69be451242a063eb2cc787a07abc42/tests/Microsoft.DotNet.XHarness.Apple.Tests/Orchestration/OrchestratorTestBase.cs) - this is common for all of these tests.

Things like `_appInstaller` setups differ in small details for every test - such as we return different the `ExitCode` or we check different parameters (`_device` vs `_simulator`), we are changing booleans and so on.
For example here we are testing the Install command - so we need different behavior of these in each setup so there is no common code to extract anymore.

Or do you have some repetitive specific code piece I didn't notice?

Reply:
Discussed synchronously, you are probably right.